### ISSUE TITLE
AngularFireModule has not been provided in your AppModule when lazy loading Firestore

### ISSUE THREAD

**Issue by MawRojas** [angular/angularfire#3404]
### Version info

**Angular:** 15.2.9

**Firebase:** 9.8  

**AngularFire:** 7.5  

### How to reproduce these conditions

Based on the [sample advanced app](https://github.com/angular/angularfire/tree/master/samples/advanced). Firestore should be able to be lazy loaded.  
I've created a getFiresotre file as the sample app indicates:  
```
    import { getFirestore } from '@angular/fire/firestore';
    export const firestore = getFirestore();
```
The file is localed in the following path: `src/app/getFirestore.ts`   
Then, I'm importing firestore just like the sample app:  
```
    import { firestore } from 'src/app/getFirestore';
```
However, the getFirestore file throws the following error: `Error: Either AngularFireModule has not been provided in your AppModule (this can be done manually or implictly using provideFirebaseApp) or you're calling an AngularFire method outside of an NgModule (which is not supported)`
The error makes sense, since I'm never providing Firestore in the app.module. However, the sample app does not provide it either.  
I have inspected the sample app for the part of the code that provides firestore; nonetheless, I was not able to find it.  
Was is the correct way of providing firestore while lazy loading it?

**Comment by davideast** [comment]
@MawRojas Can you provide an example on StackBlitz? That would make it much easier to debug.

**Comment by MawRojas** [comment]
> @MawRojas Can you provide an example on StackBlitz? That would make it much easier to debug.

Hello, I've tried to setup the StackBlitz project but I cannot get it to compile when I provide the firebase app in app.module using `provideFirebaseApp`.  
Also, the provided "Angular Fire Start" project, that shows up when submitting a new issue, is using the old angularfire version, 6, so that is not useful in this case.  
Could you tell me how to get it to compile? Once that is done I'll replicate the problem I described in this issue.  [Link to the StackBlitz project](https://stackblitz.com/edit/angular-ivy-bkk1dt?file=src%2Fapp%2Ffirestore.service.ts)

Lastly, I found another project that is setup for angularfire 7 but it also does not compile. I don't know if that is useful but [this is the link](https://stackblitz.com/edit/angular-ivy-7tndk2?file=src%2Fapp%2Fapp.module.ts)

**Comment by davideast** [comment]
@MawRojas After a bit of investigation I think I understand the problem a bit more now. I need to debug a bit more but hang tight!

**Comment by MawRojas** [comment]
> @MawRojas After a bit of investigation I think I understand the problem a bit more now. I need to debug a bit more but hang tight!

Sounds good. Thank you for looking into it.

**Comment by MawRojas** [comment]
@davideast Hello, any news regarding the lazy loading?

**Comment by armando-navarro** [comment]
Thanks for the report. This was based on the older advanced sample on @angular/fire 7.5, and both the library (standalone providers and injection contexts) and the sample have changed substantially since then, so the lazy-loading pattern you followed no longer applies. The sample-app refresh is tracked in #3669.

I'm closing this as tied to an obsolete version, but if lazy-loading Firestore still throws the provider error on a current version, please open a new issue with a minimal reproduction.
